### PR TITLE
793 > Fix Back Browser Button & Enter Key issues

### PR DESCRIPTION
**Pull request by thejoefriel**
Relates to #793 

There are two things to achieve in this fix:
1. Prevent users from going back on sign up and reviews (and losing data) when they hit back to close a popup
2. Make sure when users hit enter, done, go etc when they want to create new organisation in the search page, that this doesn't just remove the data 

On point 1, it isn't possible to prevent the browser back button. So what I've done is put in a warning message to the user that lets them know they will lose their data if they continue

### CODE REVIEW

File: client/src/Components/Common/Popover/index.js [hunk 1]
```diff
@@ -22,12 +23,31 @@ class PopoverComponent extends React.Component {
   };
 
   componentDidMount() {
+    this.setState({ currentUrl: window.location.pathname });
```
Comment (RamyAlshurafa) [inline client/src/Components/Common/Popover/index.js]: is this used?

Comment (thejoefriel) [reply client/src/Components/Common/Popover/index.js]: No you're right, I'll remove :+1:

**Comment by thejoefriel** [comment]
> Nice solution @thejoefriel, really liked it 👍
> just wanted to add this Prompt when the select is open and user selected to `create new`
> I think you should check if the add modal is opened and and if its opened then add the prompt
> in line 150 in `Common/Select` use this
> 
> ```
> onClick={()=>{
>  addHandler();
>   this.setState(isModalOpen: true);
> }
> ```
> 
> then check isModalOpen to add the promt or not
> also resolve the conflict please

Hey @RamyAlshurafa sorry I don't quite understand what you mean here. I'm fixing the merge conflict then pushing up. Did you want to make the change you are suggesting please?

**Comment by RamyAlshurafa** [comment]
Here what i mean 
Screenshot_20191212_095434_org mozilla firefox

**Comment by RamyAlshurafa** [comment]
@thejoefriel 
In the previous image, when the user wants to add new trade/entities/... should apply the prompt in this case

**Pull request merged**